src/lib.rs

```diff
@@ -316,6 +337,8 @@ pub struct GlobalVariable {
     pub binding: Option<Binding>,
     /// The type of this variable.
     pub ty: Handle<Type>,
+    /// The interpolation qualifier, if any.
```

Comment:
on IR level, we should be more strict and say that this is required for global inputs in FS and global outputs in VS

Reply:
Should I try to encode that invariant in the type system somehow? Or just add it as a comment?

Reply:
Just as a comment. I tried to think about how to encode this more, and didn't find a way.